### PR TITLE
boards: esp_wrover_kit: use gpio hogs for LCD, drop LEDs

### PR DESCRIPTION
**Pull request by gmarull**
Instead of using custom board init code, use a GPIO hog to initialize
the LCD backlight. Note that LCD backlight should be controller by the
display driver, but this is not supported now. Initializing defined LEDs
should be done by the application, using either the GPIO or LEDs driver
classes.

### CODE REVIEW

File: boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts [hunk 1]
```diff
@@ -88,6 +88,19 @@
 
 &gpio0 {
 	status = "okay";
+
+	rgb-leds-off {
+		gpio-hog;
+		gpios = <4 GPIO_ACTIVE_HIGH>, <2 GPIO_ACTIVE_HIGH>,
+			<0 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
```
Comment (fabiobaltieri) [inline boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts:97]: These are already in `gpio-leds`, how about just dropping it?

Comment (gmarull) [reply boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts:97]: they may be used later in the app, right?

Comment (fabiobaltieri) [reply boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts:97]: yeah but then the app should reinitialize them anyway or use the leds subsys

Comment (gmarull) [reply boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts:97]: so then I guess we should drop LEDs config, and just leave LCD, right?

Comment (fabiobaltieri) [reply boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts:97]: yeah I think it'd be reasonable

**Review by fabiobaltieri — DISMISSED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**